### ISSUE TITLE
Downloading first manifest... [ERROR] Downloading first manifest failed with code: 0

### ISSUE THREAD

**Issue by hellishvictor** [Data-Oriented-House/PortableBuildTools#11]
Hi, here in Win 7 x64 SP1 I get the following:
```
Downloading first manifest...
[ERROR] Downloading first manifest failed with code: 0
```
Cheers.

**Comment by j-begin** [comment]
This release has added official Windows 7 support. 

It works on my end, but please test it on yours.

**Comment by hellishvictor** [comment]
Man, it shows:
```
Downloading manifest files...
send request failed
Failed to download release manifest
```
:S

**Comment by hellishvictor** [comment]
With the [original script](https://gist.githubusercontent.com/mmozeiko/7f3162ec2988e81e56d5c4e22cde9977) the download works:
```
Downloading MSVC v14.40.17.10 and Windows SDK v26100
Do you accept Visual Studio license at https://go.microsoft.com/fwlink/?LinkId=2179911 [Y/N] ? y
microsoft.vc.14.40.17.10.tools.hostx64.targetx64.base ... 100%
microsoft.vc.14.40.17.10.tools.hostx64.targetx64.res.base ... 100%
microsoft.vc.14.40.17.10.crt.headers.base ... 100%
microsoft.vc.14.40.17.10.crt.x64.desktop.base ... 100%
microsoft.vc.14.40.17.10.crt.x64.store.base ... 100%
microsoft.vc.14.40.17.10.crt.source.base ... 100%
microsoft.vc.14.40.17.10.asan.headers.base ... 100%
microsoft.vc.14.40.17.10.asan.x64.base ... 100%
Windows SDK for Windows Store Apps Tools-x86_en-us.msi ... 100%
Windows SDK for Windows Store Apps Headers-x86_en-us.msi ... 100%
Windows SDK Desktop Headers x86-x86_en-us.msi ... 100%
Windows SDK for Windows Store Apps Libs-x86_en-us.msi ... 100%
Windows SDK Desktop Libs x64-x86_en-us.msi ... 100%
Universal CRT Headers Libraries and Sources-x86_en-us.msi ... 100%
2630bae9681db6a9f6722366f47d055c.cab ... 100%
```
etc...

**Comment by munohikari** [comment]
Which Windows 7 version do you have, RTM? Because it works on my SP1 no problem. If not that - no idea what can cause it.

**Comment by hellishvictor** [comment]
Does the program use TLS certificate verification or something or like that? [Check this out,](https://github.com/Hirevo/mega-cli-rs/issues/1#issuecomment-2016488473)

**Comment by munohikari** [comment]
@hellishvictor could you try this binary, see if it fixes the issue?
https://github.com/Data-Oriented-House/PortableBuildTools/releases/download/v2.0-pre/PortableBuildTools.exe

**Comment by hellishvictor** [comment]
It shows `Downloading manifest files...`, creates the folder _%TEMP%\BuildTools_, download a .tmp file of 5/6MiB on the _%TEMP%\_ folder, and seems to froze in there. Yesterday the py script worked like a charm and downloaded the WinSDKVersion 10.0.26100.0, I build an exe and everything was fine.

**Comment by munohikari** [comment]
I never download any `.tmp` files, only `.json` ones, nor are any of them 5/6MiB, they are either less than 1Mib or 14-15MiB. Please report the information correctly next time.

That said, it is some progress that it at least attempts to download it, i wonder where it gets stuck though. As i don't have your exact setup, and no SSL errors, i cannot reproduce the issue. You could try building it (just type `build`) and running it with a debugger, pausing when it gets stuck and see the state.

Sidenote, does 26100 `cl.exe` really work for you? I tried compiling this project on my Windows 7 machine with 26100 SDK, but it doesn't work since it is missing some header file available only since Windows 8.

**Comment by hellishvictor** [comment]
Yeah, that at least is [something](https://gyazo.com/fc937dfe364ea4eb14e6c2318c252bbe), I just thought it was part of the **vs_BuildTools** or something related to it, but anyway, I've downloaded the master of **PBT** and at build it with the update of yesterday and it showed:
```
C:\vs_BuildTools\Windows Kits\10\Include\10.0.26100.0\um\windows.h(1) : fatal error RC1015: cannot open include file 'winapifamily.h'.
source.c
C:\vs_BuildTools\Windows Kits\10\Include\10.0.26100.0\um\windows.h(1): fatal error C1083: Cannot open include file: 'winapifamily.h': No such file or directory
```
That "_winapifamily.h_" was removed from the new 26100. With the previous one it build without problem. Good to now that,

**Comment by munohikari** [comment]
Now you need to use SDK less than 26100 for Windows 7. As for the networking issue, seeing as i cannot reproduce your issue, i have no idea how to fix it either. I know the place where it loops, but to properly handle it, i would need to examine what is going on in detail, what responses i get and so on. So i'll make a special debug build for you, with a lot more logging, to see what is going on.

Please try this:
https://github.com/Data-Oriented-House/PortableBuildTools/releases/download/v2.1-pre/pbt.exe

**Comment by hellishvictor** [comment]
It shows:
```
Downloading manifest files...
LOG: send request failed, seeing if it wants to re-send it...
LOG: SSL failure occured, trying to set options to allow it...
LOG: Options set successfully.
LOG: send request failed, seeing if it wants to re-send it...
LOG: SSL failure occured, trying to set options to allow it...
LOG: Options set successfully.
LOG: send request failed, seeing if it wants to re-send it...
LOG: SSL failure occured, trying to set options to allow it...
LOG: Options set successfully.
LOG: send request failed, seeing if it wants to re-send it...
LOG: SSL failure occured, trying to set options to allow it...
LOG: Options set successfully.
LOG: send request failed, seeing if it wants to re-send it...
LOG: SSL failure occured, trying to set options to allow it...
LOG: Options set successfully.
LOG: send request failed, seeing if it wants to re-send it...
LOG: SSL failure occured, trying to set options to allow it...
LOG: Options set successfully.
LOG: send request failed, seeing if it wants to re-send it...
LOG: SSL failure occured, trying to set options to allow it...
LOG: Options set successfully.
```
And continue like that,

**Comment by munohikari** [comment]
Okay, could you try this build, see if it errors?
https://github.com/Data-Oriented-House/PortableBuildTools/releases/download/v2.1-pre/pbt.exe

**Comment by hellishvictor** [comment]
Awesome, you nailed it man!
![sshot-1](https://github.com/Data-Oriented-House/PortableBuildTools/assets/34694956/d9ed52bf-9733-4993-837e-7f04ecb28317)
That's is my temp folder with the jsons it creates. Don't know how you did it, but it worked like a charm and very fast.
Later I'll try to download a sdk version with it and test how it goes. どうもありがとうございます :

**Comment by munohikari** [comment]
@hellishvictor btw i was wrong, it's not an issue with 26100 on Windows 7, it was an issue with the SDK itself. New version should install it properly:
https://github.com/Data-Oriented-House/PortableBuildTools/releases/tag/v2.2-pre

**Comment by hellishvictor** [comment]
`Success! PortableBuildTools was installed successfully.`

It build and worked great, thank you man!

**Comment by RealYukiSan** [comment]
the installation program didn't work on both v2.2 and v2.6 (the currently latest version)
![image](https://github.com/user-attachments/assets/cfa5b61b-2f12-4f83-bffc-5bce3bae0e9b)

**Comment by munohikari** [comment]
Perhaps you need Pro edition, i only tested it on Pro.

**Comment by RealYukiSan** [comment]
same error on Pro version

![image](https://github.com/user-attachments/assets/61009837-ece9-4fdd-92b9-7f5eed6343e4)
